src/umpire/Umpire.cpp

```diff
@@ -253,4 +255,32 @@ MPI_Comm get_communicator_for_allocator(Allocator a, MPI_Comm comm)
 }
 #endif
 
+bool try_coalesce(Allocator a)
+{
+  auto s = a.getAllocationStrategy();
+  bool coalesced{false};
+
+  strategy::QuickPool* qp{dynamic_cast<strategy::QuickPool*>(s)};
+  if (qp) {
+    qp->coalesce();
+    coalesced = true;
+  }
+
+  strategy::DynamicPoolList* dpl{dynamic_cast<strategy::DynamicPoolList*>(s)};
```

Comment:
Should this be placed within an `else {}` block for the case when `qp` is a `nullptr`?